index.mld

```diff
@@ -0,0 +1,97 @@
+- [-config=[FILE]] Use [[FILE]] as the configuration file.
+If you omit this option, {i cwe_checker} uses a standard configuration file located at [src/config.json].
+- [-module-versions] Prints the version numbers of each check.
+- [-json] Format the CWE-warnings as JSON.
+If you print to {i stdout}, note that debug, info and error messages are not formatted as JSON and may pollute the output.
+Use [-no-logging] to suppress these messages.
+- [-no-logging] Suppress printing of debug, info and error messages.
+This is useful if you want to print to {i stdout} with the [-json] flag to prevent these messages polluting the JSON output.
+- [-out=[FILE]] Print the CWE-warnings to the file located at [[FILE]].
+Note that debug, info and error messages are still printed to stdout and not to the file.
+- [-partial=[MODULE_LIST]] Only run the checks given in [[MODULE_LIST]], where [[MODULE_LIST]] is a comma separated list of module names.
+E.g. [-partial=CWE190,CWE476] would only run the checks for CWE-190 and CWE-476.
+The names of all available modules can be printed with the [-module-versions] command line option.
+
+{1:ToolIntegration Integration with IDA Pro}
+
+To annotate CWE-hits in IDA Pro, first run {i cwe_checker} and save the JSON-formatted output to a file.
+{[bap [BINARY] --pass=cwe-checker --cwe-checker-json --cwe-checker-out=cwe_hits.json]}
+Then run the [cwe_checker_to_ida.py] script located in the [cwe_checker_to_ida] folder.
+{[python3 cwe_checker_to_ida.py -i cwe_hits.json -o cwe_hits.py]}
+Now open the binary file in IDA Pro and execute the generated [cwe_hits.py] script from within IDA Pro (Alt+F7).
+
+{1:HackingHowto How to write your own check}
+
+{2 Step 1: Get acquainted with BAP}
+
+{i cwe_checker} uses the {{: https://github.com/BinaryAnalysisPlatform/bap} Binary Analysis Platform} as a backend and you should read its documentation.
+All information about a binary file is gathered through BAP.
+
+{2 Step 2: Write the check}
+
+You need a [your_check.ml] and a [your_check.mli] file that should be located in the [src/checkers] folder.
+The [your_check.mli] file should look like this:
+{[
+(** This module implements your_check.
+
+    Some more documentation about your_check.
+*)
+
+val name : string     (* The name of your check *)
+val version : string  (* The version of your check (e.g. "0.1"). *)
+
+val check_cwe : Bap.Std.program Bap.Std.term -> Bap.Std.project -> Bap.Std.word Bap.Std.Tid.Map.t -> string list list -> string list -> unit
+]}
+The {i check_cwe} function has as parameters:
+- [program: Bap.Std.program Bap.Std.term] The program term of the binary
+- [project: Bap.Std.project] The BAP-project term of the binary
+- [Tid_map: Bap.Std.word Bap.Std.Tid.Map.t] A map from the Tids of basic blocks to concrete addresses in the binary file
+- [symbols_pairs: string list list] Symbols read from the {i config.json} file
+- [parameters: string list] Parameters read from the config.json file
+
```

Comment:
story: add an empty check that just outputs "Hello, World!"

Reply:
I have added an example .ml to the documentation, but I am not eager to add an actual hello-world-check to the code base. I hope the documentation is already clear enough so that users can copy-paste this example to the code base.